File: reverse_words.py
# Complete the solution so that it reverses all of the words within the string passed in.

# Example:

# reverseWords("The greatest victory is that which requires no battle")
# // should return "battle no requires which that is victory greatest The"

def reverseWords(s):
    reverseString = s.split() #split the words into a list: ['hello', 'world!']
    reverseString.reverse() #reverse the list: ['world!', 'hello']
    return " ".join(reverseString) # join back into string: 'world! hello'

s = "hello world!"

print(reverseWords(s))


    # CodeWars Solution:
    # def reverseWords(str):
    #     return " ".join(str.split(" ")[::-1])

File: reverseWords.ts
// Complete the function that accepts a string parameter, and reverses each word in the string. All spaces in the string should be retained.

// Examples
// "This is an example!" ==> "sihT si na !elpmaxe"
// "double  spaces"      ==> "elbuod  secaps"

export function reverseWords(str: string): string {
    let sentenceString: string = str;
    let stringArray: string[] = sentenceString.split(' ');
    let arrayLength = stringArray.length;
    
    let newString: string = ''

    for (let i = 0; i < arrayLength; i++) {
        let wordString: string = stringArray[i]
        let wordArray = wordString.split('')
        newString += wordArray.reverse().join('') + ' ';
    }

    return newString
}

console.log(reverseWords('The quick brown fox jumps over the lazy dog.'));
console.log(reverseWords('apple'));
console.log(reverseWords('a b c d'));
console.log(reverseWords('double  spaced  words'));